blockchain/consensus/deleg_cns/src/proposer.rs

```diff
@@ -102,18 +102,18 @@ impl Proposer for DelegatedProposer {
         state_manager: Arc<StateManager<DB>>,
         mpool: Arc<MP>,
         submitter: SyncGossipSubmitter,
-        services: &mut JoinSet<()>,
+        services: &mut JoinSet<anyhow::Result<()>>,
     ) -> anyhow::Result<()>
     where
         DB: Blockstore + Store + Clone + Sync + Send + 'static,
         MP: MessagePoolApi + Send + Sync + 'static,
     {
         services.spawn(async move {
-            if let Err(e) = self.run(state_manager, mpool.as_ref(), &submitter).await {
-                error!("block proposal stopped: {}", e)
-            }
+            self.run(state_manager, mpool.as_ref(), &submitter)
+                .await
+                .map_err(|e| anyhow::anyhow!("block proposal stopped: {}", e))
         });
-        Ok(())
+        anyhow::Ok(())
```

Comment:
Is this necessary?

Reply:
Yes, as `Proposer` trait requires `anyhow::Result` as return type